pkg/sql/create_type.go

```diff
@@ -78,6 +78,11 @@ func resolveNewTypeName(
 func getCreateTypeParams(
 	params runParams, name *tree.TypeName, db *sqlbase.ImmutableDatabaseDescriptor,
 ) (typeKey sqlbase.DescriptorKey, schemaID sqlbase.ID, err error) {
+	if name.Schema() == tree.PublicSchema {
+		if _, ok := types.PublicSchemaAliases[name.Object()]; ok {
```

Comment:
fancy

need to also do the same in getCreateTableParams

Reply:
we can already add tables with type names atm (though i'm sure that's not intentional)

```
root@127.0.0.1:59943/defaultdb> create table int4 (a int);
CREATE TABLE

Time: 7.211ms
```

i'd rather save that change for something more wide-catching.

Reply:
ill add a TODO i guess :P

Reply:
ah nvm i'm dumb!